### ISSUE TITLE
renv::status() error reporting "regular expression too large".

### ISSUE THREAD

**Issue by karimn** [rstudio/renv#652]
Hi,

I'm using {renv} version 0.12.5. Things were running fine till today when I started seeing the below error. The only recent change is installing a package using `devtools::install_local` to debug some issues. After that, I reinstalled the package from github but the problem didn't go away.

```
> renv::status()
Error in grepl(patterns$exclude %||% "", children, perl = TRUE) : 
  invalid regular expression '(?:^\Q/media/karim/Code Drive/karimn-code/nls-rct-2017-2019/bangladesh_rainfall/data/newdata/\E(?:(?:.*/)?\Q20070101.nc\E$|(?:.*/)?\Q20070102.nc\E$|(?:.*/)?\Q20070103.nc\E$|(?:.*/)?\Q20070104.nc\E$|(?:.*/)?\Q20070105.nc\E$|(?:.*/)?\Q20070106.nc\E$|(?:.*/)?\Q20070107.nc\E$|(?:.*/)?\Q20070108.nc\E$|(?:.*/)?\Q20070109.nc\E$|(?:.*/)?\Q20070110.nc\E$|(?:.*/)?\Q20070111.nc\E$|(?:.*/)?\Q20070112.nc\E$|(?:.*/)?\Q20070113.nc\E$|(?:.*/)?\Q20070114.nc\E$|(?:.*/)?\Q20070115.nc\E$|(?:.*/)?\Q20070116.nc\E$|(?:.*/)?\Q20070117.nc\E$|(?:.*/)?\Q20070118.nc\E$|(?:.*/)?\Q20070119.nc\E$|(?:.*/)?\Q20070120.nc\E$|(?:.*/)?\Q20070121.nc\E$|(?:.*/)?\Q20070122.nc\E$|(?:.*/)?\Q20070123.nc\E$|(?:.*/)?\Q20070124.nc\E$|(?:.*/)?\Q20070125.nc\E$|(?:.*/)?\Q20070126.nc\E$|(?:.*/)?\Q20070127.nc\E$|(?:.*/)?\Q20070128.nc\E$|(?:.*/)?\Q20070129.nc\E$|(?:.*/)?\Q20070130.nc\E$|(?:.*/)?\Q20070131.nc\E$|(?:.*/)?\Q20070201.nc\E$|(?:.*/)?\Q20070202.nc\E$|(?:.*/)?\Q20070203.nc\E$|(?:.*/)?\Q2007
In addition: Warning message:
In grepl(patterns$exclude %||% "", children, perl = TRUE) :
  PCRE pattern compilation error
	'regular expression is too large'
	at ''
Traceback (most recent calls last):
38: renv::status()
37: renv_status_impl(project, libpaths, lockpath, cache)
36: renv_status_check_missing_library(project, libpaths)
35: snapshot(project = project, library = libpaths, lockfile = NULL, 
        force = TRUE)
34: renv_lockfile_create(project, libpaths, type, packages)
33: renv_snapshot_r_packages(libpaths = libpaths, project = project) %>% 
        renv_snapshot_filter(project = project, type = type, packages = packages) %>% 
        renv_snapshot_fixup()
32: eval(call, envir = parent.frame())
31: eval(call, envir = parent.frame())
30: renv_snapshot_fixup(renv_snapshot_r_packages(libpaths = libpaths, 
        project = project) %>% renv_snapshot_filter(project = project, 
        type = type, packages = packages))
29: renv_snapshot_fixup_renv(records)
28: renv_snapshot_r_packages(libpaths = libpaths, project = project) %>% 
        renv_snapshot_filter(project = project, type = type, packages = packages)
27: eval(call, envir = parent.frame())
26: eval(call, envir = parent.frame())
25: renv_snapshot_filter(renv_snapshot_r_packages(libpaths = libpaths, 
        project = project), project = project, type = type, packages = packages)
24: renv_snapshot_filter_implicit(project, records)
23: renv_snapshot_filter_impl(project, records, project)
22: renv_snapshot_dependencies(project, source)
21: withCallingHandlers(dependencies(path = source, root = project, 
        progress = FALSE, errors = errors), renv.dependencies.error = renv_dependencies_error_handler(message, 
        errors))
20: dependencies(path = source, root = project, progress = FALSE, 
        errors = errors)
19: delegate(renv_dependencies_impl)
18: eval(call, envir = parent.frame(2))
17: eval(call, envir = parent.frame(2))
16: <condition-handler>(...)
15: renv_dependencies_find(path, root)
14: lapply(path, renv_dependencies_find_impl, root = root)
13: FUN(X[[i]], ...)
12: renv_dependencies_find_dir(path, root)
11: lapply(children, renv_dependencies_find_impl, root = root)
10: FUN(X[[i]], ...)
 9: renv_dependencies_find_dir(path, root)
 8: lapply(children, renv_dependencies_find_impl, root = root)
 7: FUN(X[[i]], ...)
 6: renv_dependencies_find_dir(path, root)
 5: lapply(children, renv_dependencies_find_impl, root = root)
 4: FUN(X[[i]], ...)
 3: renv_dependencies_find_dir(path, root)
 2: renv_renvignore_exec(path, root, path)
 1: grepl(patterns$exclude %||% "", children, perl = TRUE)
```

**Comment by kevinushey** [comment]
Did you change your `.gitignore` recently? `renv` reads and uses that when trying to figure out what files to "ignore" in a project; I think we're inadvertently blowing up the regex engine with a too-large regex generated from that file.

**Comment by kevinushey** [comment]
I've just pushed a potential fix: can you try installing the latest development version of `renv` and let me know if the issue still persists for you?

**Comment by karimn** [comment]
No, I haven't changed anything with `.gitignore` recently. 

Do I install the latest dev version using `devtools::install_github("rstudio/renv")` or is there some branch I need to specify?

**Comment by kevinushey** [comment]
That's correct.

**Comment by karimn** [comment]
Yes, that seems to fix it (version 0.12.5-59). Thanks!

**Comment by kevinushey** [comment]
Awesome! Thank you for verifying; I'll hopefully have an updated version of `renv` on CRAN soon.
